### PR TITLE
support multi metrics

### PR DESCRIPTION
**Pull request by Guybenisrael**

### CODE REVIEW

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
+type: object
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:1]: Use camel-case for file name

File: api/services/elastigroup/aws/schemas/elastigroup-scale-up.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
+type: object
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-up.yaml:1]: Use camel-case for file name

File: api/services/elastigroup/aws/schemas/elastigroup-scaling.yaml [hunk 3]
```diff
@@ -2,283 +2,25 @@ type: object
 title: Elastigroup Scaling
 properties:
   up:
+    type: array
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scaling.yaml:5]: Use camel-case for file name

File: api/services/elastigroup/aws/schemas/elastigroup-target-scale.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
+type: object
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-target-scale.yaml:1]: Use camel-case for file name

File: api/services/elastigroup/aws/schemas/multimetric-expressionconfig.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+type: object
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/multimetric-expressionconfig.yaml:1]: Use camel-case for file name, and use: multipleMetrics

File: api/services/elastigroup/aws/schemas/multimetric-metricconfig.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
+type: object
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/multimetric-metricconfig.yaml:1]: Use camel-case for file name, and use: multipleMetrics

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+title: scale down config
+properties:
+  policyName:
+    type: string
+    description: |
+      The policy name
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:7]: End sentence with dot (.)

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+  statistic:
+    type: string
+    description: |
+      The metric statistics to return.
+    enum: [ average , sum , sampleCount , maximum , minimum , percentile ]
+    example: "average"
+  extendedStatistic:
+    type: string
+    description: |
+      Percentile statistic. Valid values: p0.1 - p100
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:24]: End sentence with dot (.)

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+  unit:
+    type: string
+    description: |
+      The unit for the alarm's associated metric.
+    enum: [ seconds , microseconds , milliseconds , bytes , kilobytes , megabytes , gigabytes , terabytes , bits , kilobits , megabits , gigabits , terabits , percent , count , bytes/second , kilobytes/second , megabytes/second , gigabytes/second , terabytes/second , bits/second , kilobits/second , megabits/second , gigabits/second , terabits/second , count/second , none ]
+    example: "percent"
+  threshold:
+    type: integer
+    description: |
+      The value against which the specified statistic is compared
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:35]: End sentence with dot (.)

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+    example: 90
+  action:
+    type: object
+    example: "adjustment"
+    description: |
+      The action to take when scale up is needed.
+      A mandatory field, that should be filled with one of the following:
+      "adjustment", "percentageAdjustment", "setMaxTarget", "setMinTarget", "updateCapacity".
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:43]: why this is not enum ?

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+      The unit for the alarm's associated metric.
+    enum: [ seconds , microseconds , milliseconds , bytes , kilobytes , megabytes , gigabytes , terabytes , bits , kilobits , megabits , gigabits , terabits , percent , count , bytes/second , kilobytes/second , megabytes/second , gigabytes/second , terabytes/second , bits/second , kilobits/second , megabits/second , gigabits/second , terabits/second , count/second , none ]
+    example: "percent"
+  threshold:
+    type: integer
+    description: |
+      The value against which the specified statistic is compared
+    example: 90
+  action:
+    type: object
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:38]: why this is not string?

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+    example: "adjustment"
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:39]: This is an example for the whole action object..? it is wrongly displayed on the client

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+        type: string
+        description: |
+          The type of the action to take when scale up is needed.
+        enum: [ adjustment , updateCapacity setMinTarget , percentageAdjustment ]
+        example: "updateCapacity"
+      adjustment:
+        type: integer
+        description: |
+          The number / percentage associated with the specified adjustment type.
+          Required if using `adjustment` or `percentageAdjustment` as action type
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:55]: End sentence with a dot (.)

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+        example: 1
+      minTargetCapacity:
+        type: integer
+        description: |
+          The number with the target capacity needed.
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:60]: "The minimum target capacity."

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+          Required if using `setMinTarget` as action type
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:61]: End sentence with a dot (.)

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+        example: 1
+      target:
+        type: integer
+        description: |
+          The desired number of instances. Required if using `updateCapacity` as action type and neither `minimum` nor `maximum` are not defined.
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:66]: "and neither `minimum` nor `maximum` are defined."

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+        example: 1
+      minimum:
+        type: integer
+        description: |
+          The lower limit number of instances that you can scale down to. Optional, required if using `updateCapacity` as action type and neither `target` nor `maximum` are not defined
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:71]: "and neither `target` nor `maximum` are defined."

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+        example: 0
+      maximum:
+        type: integer
+        description: |
+          The upper limit number of instances that you can scale up to. Optional, required if using `updateCapacity` as action type and neither `target` nor `minimum` are not defined
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:76]: "and neither `target` nor `minimum` are defined."

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+  namespace:
+    type: string
+    description: |
+      The namespace for the alarm's associated metric.
+    default: "AWS/EC2"
+  dimensions:
+    type: array
+    description: |
+      The dimensions for the alarm's associated metric.
+      If the user mentioned name as instanceId, there is NO value
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:87]: This sentence is not clear. Where there is no value? also, no need to use the word 'user'

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml [hunk 20]
```diff
@@ -0,0 +1,120 @@
...
+    example: 300
+  evaluationPeriods:
+    type: integer
+    description: |
+      The number of periods over which data is compared to the specified threshold.
+    example: 1
+  cooldown:
+    type: integer
+    description: |
+      The amount of time, in seconds, after a scaling activity completes before any further trigger-related scaling activities can start.
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:107]: "The amount of time, in seconds, after a scaling activity completes and before..."

File: api/services/elastigroup/aws/schemas/elastigroup-target-scale.yaml [hunk 21]
```diff
@@ -0,0 +1,46 @@
...
+title: target scaling config
+properties:
+  source:
+    title: Source
+    description: "The source of the metric Supported values: “cloudWatch”, “spectrum”"
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-target-scale.yaml:6]: New line needed after 'metric'

File: api/services/elastigroup/aws/schemas/elastigroup-target-scale.yaml [hunk 22]
```diff
@@ -0,0 +1,46 @@
...
+    type: string
+    example: spectrum
+  policyName:
+    type: string
+    description: |
+      The policy name
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-target-scale.yaml:12]: Add dot (.) at the end of the sentence

File: api/services/elastigroup/aws/schemas/elastigroup-target-scale.yaml [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+    example: "percent"
+  namespace:
+    type: string
+    description: |
+      The namespace for the alarm's associated metric.
+    default: "AWS/EC2"
+  target:
+    type: integer
+    description: |
+      The target value for the group
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-target-scale.yaml:40]: Add dot (.) at the end of the sentence.

File: api/services/elastigroup/aws/schemas/elastigroup-target-scale.yaml [hunk 24]
```diff
@@ -0,0 +1,46 @@
...
+    example: 1
+  cooldown:
+    type: integer
+    description: |
+      The amount of time, in seconds, after a scaling activity completes before any further trigger-related scaling activities can start.
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/elastigroup-target-scale.yaml:45]: "The amount of time, in seconds, after a scaling activity completes and before..."

File: api/services/elastigroup/aws/schemas/multimetric-expressionconfig.yaml [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+title: Expression config
+required:
+  - name
+  - expression
+properties:
+  name:
+    type: string
+    description: |
+      The expression name
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/multimetric-expressionconfig.yaml:10]: Add dot (.) at the end of the sentence.

File: api/services/elastigroup/aws/schemas/multimetric-expressionconfig.yaml [hunk 26]
```diff
@@ -0,0 +1,17 @@
...
+    example: "e1"
+  expression:
+    type: string
+    description: |
+      An expression consisting of the metrics above
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/multimetric-expressionconfig.yaml:15]: Add dot (.) at the end of the sentence.

File: api/services/elastigroup/aws/schemas/multimetric-metricconfig.yaml [hunk 27]
```diff
@@ -0,0 +1,55 @@
...
+required:
+  - name
+  - metricName
+  - namespace
+properties:
+  name:
+    type: string
+    description: |
+      The policy name
+    example: "policyName"
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/multimetric-metricconfig.yaml:12]: Example should be a metric name, not policy name (better use the same example you use in the expression config - metric1)

File: api/services/elastigroup/aws/schemas/multimetric-metricconfig.yaml [hunk 28]
```diff
@@ -0,0 +1,55 @@
...
+    type: string
+    description: |
+      The unit for the alarm's associated metric.
+    enum: [ seconds , microseconds , milliseconds , bytes , kilobytes , megabytes , gigabytes , terabytes , bits , kilobits , megabits , gigabits , terabits , percent , count , bytes/second , kilobytes/second , megabytes/second , gigabytes/second , terabytes/second , bits/second , kilobits/second , megabits/second , gigabits/second , terabits/second , count/second , none ]
+    example: "percent"
+  dimensions:
+    type: array
+    description: |
+      The dimensions for the alarm's associated metric.
+      If the user mentioned name as instanceId, there is NO value
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/multimetric-metricconfig.yaml:44]: Sentence is not clear

File: api/services/elastigroup/aws/schemas/multimetric-metricconfig.yaml [hunk 29]
```diff
@@ -0,0 +1,55 @@
...
+  statistic:
+    type: string
+    description: |
+      The metric statistics to return.
+    enum: [ average , sum , sampleCount , maximum , minimum , percentile ]
+    example: "average"
+  extendedStatistic:
+    type: string
+    description: |
+      Percentile statistic. Valid values: p0.1 - p100
```
Comment (idohas) [inline api/services/elastigroup/aws/schemas/multimetric-metricconfig.yaml:32]: Add dot (.) at the end of the sentence.

File: api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml (hunk 10 above)
Comment (Guybenisrael) [reply api/services/elastigroup/aws/schemas/elastigroup-scale-down.yaml:43]: this is how it was implemented, just a copy paste.

**Review by idohas — APPROVED** [review]

**Pull request merged**